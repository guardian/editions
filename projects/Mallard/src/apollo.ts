/**
 * Setup Apollo framework. At the moment we use Apollo Local State management
 * mechanism and don't fetch data from GraphQL endpoints.
 */
import { ApolloClient } from 'apollo-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { SETTINGS_RESOLVERS } from './helpers/settings/resolvers'
import { resolveWeather } from './helpers/location'

/**
 * Resolvers is what Apollo uses to get the value of field that has never been
 * cached before. For example if you have a query `{ foo @client }`, Apollo
 * looks for the field `foo` at the root of the cache. If not available, it'll
 * look for the resolver `Query.foo` (`Query` is just the "namespace" where
 * all queries start).
 *  *
 * Resolvers are functions, which can be asynchonous. They can gather data from
 * local storage (ex. settings), from local APIs (ex. permissions, geolocation),
 * from remote APIs (ex. REST API).
 *
 * Once a resolver returns a value, Apollo will refresh automatically
 * any component that was requiring this in a query via the `useQuery` function.
 */
const RESOLVERS = {
    Query: {
        ...SETTINGS_RESOLVERS,
        weather: resolveWeather,
    },
}

/**
 * The `Link` is the interface Apollo uses to reach GraphQL servers. Since we
 * don't fetch anything from GraphQL endpoints at the moment, we use a no-op
 * `Link` object. We *must* throw to prevent bad queries from carrying through.
 *
 * Apollo would try to use the `Link` if you have a query such as `{ foo }`
 * without the `@client` annotation. In our current use case, `@client` always
 * needs to be specified.
 */
const link = {
    split: () => {
        throw new Error('Not implemented. Did you forget to use `@client`?')
    },
    concat: () => {
        throw new Error('not implemented Did you forget to use `@client`?')
    },
    request: () => {
        throw new Error('not implemented Did you forget to use `@client`?')
    },
}

export const createApolloClient = () =>
    new ApolloClient({
        cache: new InMemoryCache(),
        link,
        resolvers: RESOLVERS,
    })
