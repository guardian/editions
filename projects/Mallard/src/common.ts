export * from '../../Apps/common/src/index';
