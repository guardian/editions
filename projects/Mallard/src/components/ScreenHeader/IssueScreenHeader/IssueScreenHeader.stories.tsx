import { color, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import type { IssueOrigin } from '../../../../../Apps/common/src';
import { IssueScreenHeader } from './IssueScreenHeader';

const issue = {
	name: 'Daily Edition',
	date: '2020-06-25',
	key: 'daily-edition/2020-06-25',
	publishedId: 'daily-edition/2020-06-25/2020-06-25T00:58:19.4Z',
	localId: 'daily-edition/2020-06-25',
	fronts: [],
	origin: 'api' as IssueOrigin,
};

storiesOf('IssueScreenHeader', module)
	.addDecorator(withKnobs)
	.add('Default', () => <IssueScreenHeader />)
	.add('With Issue', () => <IssueScreenHeader issue={issue} />)
	.add('With Issue and Header Styles', () => (
		<IssueScreenHeader
			issue={issue}
			headerStyles={{
				backgroundColor: color('Background Colour', '#7D0068'),
				textColorPrimary: color('Text Colour Primary', '#007ABC'),
				textColorSecondary: color('Text Colour Secondary', '#F3C100'),
			}}
		/>
	));
