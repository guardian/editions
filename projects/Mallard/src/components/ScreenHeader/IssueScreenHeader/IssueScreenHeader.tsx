import { useNavigation } from '@react-navigation/native';
import React, { useCallback } from 'react';
import type { IssueWithFronts, SpecialEditionHeaderStyles } from 'src/common';
import { IssueTitle } from 'src/components/issue/issue-title';
import { Header } from 'src/components/layout/header/header';
import { styles } from 'src/components/styled-text';
import { logEvent } from 'src/helpers/analytics';
import { useIssueDate } from 'src/helpers/issues';
import { useEditions } from 'src/hooks/use-edition-provider';
import { RouteNames } from 'src/navigation/NavigationModels';
import { remoteConfigService } from 'src/services/remote-config';
import { IssueMenuButton } from '../../Button/IssueMenuButton';
import { EditionsMenuButton } from '../../EditionsMenu/EditionsMenuButton/EditionsMenuButton';

interface Titles {
	title: string;
	subTitle: string;
	titleStyle: any;
}

const IssueScreenHeader = React.memo(
	({
		headerStyles,
		issue,
	}: {
		headerStyles?: SpecialEditionHeaderStyles;
		issue?: IssueWithFronts;
	}) => {
		const { navigate } = useNavigation();
		const { date, weekday } = useIssueDate(issue);
		const { setNewEditionSeen, selectedEdition } = useEditions();

		const getDateString = useCallback(() => {
			const abbreviatedDay = weekday.substring(0, 3);
			return `${abbreviatedDay} ${date}`;
		}, [date, weekday]);

		const goToIssueList = useCallback(() => {
			navigate(RouteNames.IssueList);
		}, [navigate]);

		const handleEditionMenuPress = useCallback(() => {
			setNewEditionSeen();
			navigate(RouteNames.EditionsMenu);
			logEvent({
				name: 'editions_menu_button',
				value: 'editions_menu_button_opened',
			});
		}, [setNewEditionSeen, navigate]);

		const isSpecialEdition = (editionType: string) => {
			return editionType === 'Special';
		};

		const getTitles = (): Titles => {
			if (isSpecialEdition(selectedEdition.editionType)) {
				const splitTitle = selectedEdition.title.split('\n');
				return {
					title: splitTitle[0],
					subTitle: splitTitle[1],
					titleStyle: styles.issueHeavyText,
				};
			}
			const dateString = getDateString();
			return {
				title: selectedEdition.title,
				subTitle: dateString,
				titleStyle: styles.issueLightText,
			};
		};

		const { title, subTitle, titleStyle } = getTitles();

		const isEditionsMenuEnabled = remoteConfigService.getBoolean(
			'is_editions_menu_enabled',
		);

		return (
			<Header
				onPress={goToIssueList}
				action={<IssueMenuButton onPress={goToIssueList} />}
				leftAction={
					<EditionsMenuButton
						disabled={!isEditionsMenuEnabled}
						onPress={handleEditionMenuPress}
					/>
				}
				headerStyles={headerStyles}
			>
				{title ? (
					<IssueTitle
						title={title}
						subtitle={subTitle}
						titleStyle={titleStyle}
						overwriteStyles={headerStyles}
					/>
				) : null}
			</Header>
		);
	},
);

export { IssueScreenHeader };
