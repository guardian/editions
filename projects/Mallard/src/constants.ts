import { Platform } from 'react-native';
import Config from 'react-native-config';

const {
	ID_API_URL,
	MEMBERS_DATA_API_URL,
	ID_ACCESS_TOKEN,
	ITUNES_CONNECT_SHARED_SECRET,
	ANDROID_RELEASE_STREAM,
} = Config;

const FACEBOOK_CLIENT_ID = '528503751025697';

const AUTH_TTL = 86400000; // ms in a day

const CAS_ENDPOINT_URL = 'https://content-auth.guardian.co.uk/subs';

const GOOGLE_CLIENT_ID =
	Platform.OS === 'android'
		? __DEV__
			? '774465807556-5oq58e8pnfqn9sptsivri5kcs9mthef9'
			: '774465807556-cjat38acttpl7md4nfc4pfediouh7v97'
		: '774465807556-kgaj5an4pc4fmr3svp5nfpulekc1rl3n';

const LEGACY_SUBSCRIBER_ID_USER_DEFAULT_KEY = 'printSubscriberID';
const LEGACY_SUBSCRIBER_POSTCODE_USER_DEFAULT_KEY = 'printSubscriberPostcode';

const JOIN_BETA_LINK = Platform.select({
	ios: 'https://testflight.apple.com/join/O2EojUEl',
	android: 'https://play.google.com/apps/testing/com.guardian.editions',
});

export {
	CAS_ENDPOINT_URL,
	ID_API_URL,
	MEMBERS_DATA_API_URL,
	ID_ACCESS_TOKEN,
	FACEBOOK_CLIENT_ID,
	GOOGLE_CLIENT_ID,
	AUTH_TTL,
	LEGACY_SUBSCRIBER_ID_USER_DEFAULT_KEY,
	LEGACY_SUBSCRIBER_POSTCODE_USER_DEFAULT_KEY,
	ITUNES_CONNECT_SHARED_SECRET,
	ANDROID_RELEASE_STREAM,
	JOIN_BETA_LINK,
};
