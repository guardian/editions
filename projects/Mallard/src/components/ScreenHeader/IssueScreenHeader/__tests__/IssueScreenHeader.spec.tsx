import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { IssueScreenHeader } from '../IssueScreenHeader'
import { IssueOrigin } from '../../../../../../Apps/common/src'

jest.mock('react-navigation', () => ({
    withNavigation: (child: any) => child,
}))

const issue = {
    name: 'Daily Edition',
    date: '2020-06-25',
    key: 'daily-edition/2020-06-25',
    publishedId: 'daily-edition/2020-06-25/2020-06-25T00:58:19.4Z',
    localId: 'daily-edition/2020-06-25',
    fronts: [],
    origin: 'api' as IssueOrigin,
}

describe('IssueScreenHeader', () => {
    it('should match the default style', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <IssueScreenHeader issue={issue} />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should match the altered style by the prop headerStyles', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <IssueScreenHeader
                issue={issue}
                headerStyles={{
                    backgroundColor: '#7D0068',
                    textColorPrimary: '#007ABC',
                    textColorSecondary: '#F3C100',
                }}
            />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
