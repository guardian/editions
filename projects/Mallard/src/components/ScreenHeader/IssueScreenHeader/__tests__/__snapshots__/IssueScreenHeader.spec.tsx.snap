// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueScreenHeader should match the altered style by the prop headerStyles 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#052962",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingVertical": 10,
      },
      Object {
        "backgroundColor": "#7D0068",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-end",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": "100%",
        },
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "height": 49,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <View
        style={
          Object {
            "paddingLeft": 14,
          }
        }
      >
        <View
          accessibilityLabel="Regions and specials editions menu"
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 24,
              "height": 42,
              "justifyContent": "center",
              "opacity": 1,
              "width": 42,
            }
          }
        >
          <RNSVGSvgView
            bbHeight={42}
            bbWidth={42}
            fill="none"
            focusable={false}
            height={42}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                undefined,
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 42,
                  "width": 42,
                },
              ]
            }
            width={42}
          >
            <RNSVGGroup
              fill={null}
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGRect
                fill={
                  Array [
                    0,
                    4278190080,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                height="41"
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "stroke",
                  ]
                }
                rotation={0}
                rx="20.5"
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={
                  Array [
                    0,
                    4294967295,
                  ]
                }
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                width="41"
                x="0.5"
                y="0.5"
              />
              <RNSVGCircle
                cx="29"
                cy="14"
                fill={
                  Array [
                    0,
                    4294949576,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                r="7"
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
              <RNSVGCircle
                cx="14"
                cy="14"
                fill={
                  Array [
                    0,
                    4294967295,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                r="7"
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
              <RNSVGCircle
                cx="29"
                cy="29"
                fill={
                  Array [
                    0,
                    4287683839,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                r="7"
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
              <RNSVGCircle
                cx="14"
                cy="29"
                fill={
                  Array [
                    0,
                    4294934287,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                r="7"
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 0,
            "flexDirection": "row",
          },
          Object {
            "width": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "flexGrow": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexGrow": 1,
              "flexShrink": 0,
              "paddingHorizontal": 14,
            }
          }
        >
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "flexShrink": 0,
                        "fontFamily": "GTGuardianTitlepiece-Bold",
                        "fontSize": 24,
                        "lineHeight": 26,
                      },
                      Array [
                        Object {
                          "color": "#ffffff",
                          "marginTop": -2,
                        },
                        undefined,
                        Object {
                          "color": "#007ABC",
                        },
                      ],
                    ]
                  }
                >
                  Thursday
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "flexShrink": 0,
                        "fontFamily": "GTGuardianTitlepiece-Bold",
                        "fontSize": 24,
                        "lineHeight": 26,
                      },
                      Array [
                        Object {
                          "color": "#ffffff",
                          "marginTop": -2,
                        },
                        Object {
                          "color": "#ffe500",
                        },
                        Object {
                          "color": "#F3C100",
                        },
                      ],
                    ]
                  }
                >
                  25 June
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
        <View
          style={
            Object {
              "paddingRight": 14,
            }
          }
        >
          <View
            accessibilityHint="More issues"
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 999,
                    "height": 42,
                    "justifyContent": "center",
                    "paddingHorizontal": 21,
                  },
                  Object {
                    "backgroundColor": "#ffe500",
                  },
                  Object {
                    "alignItems": "center",
                    "aspectRatio": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 0,
                    "width": 42,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "fontFamily": "GuardianIcons-Regular",
                      "fontSize": 20,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "color": "#121212",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`IssueScreenHeader should match the default style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#052962",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-end",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": "100%",
        },
        Array [
          Object {
            "marginTop": 20,
          },
          Object {
            "height": 49,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <View
        style={
          Object {
            "paddingLeft": 14,
          }
        }
      >
        <View
          accessibilityLabel="Regions and specials editions menu"
          accessibilityRole="button"
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 24,
              "height": 42,
              "justifyContent": "center",
              "opacity": 1,
              "width": 42,
            }
          }
        >
          <RNSVGSvgView
            bbHeight={42}
            bbWidth={42}
            fill="none"
            focusable={false}
            height={42}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                undefined,
                Object {
                  "opacity": 1,
                },
                Object {
                  "flex": 0,
                  "height": 42,
                  "width": 42,
                },
              ]
            }
            width={42}
          >
            <RNSVGGroup
              fill={null}
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGRect
                fill={
                  Array [
                    0,
                    4278190080,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                height="41"
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "stroke",
                  ]
                }
                rotation={0}
                rx="20.5"
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={
                  Array [
                    0,
                    4294967295,
                  ]
                }
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                width="41"
                x="0.5"
                y="0.5"
              />
              <RNSVGCircle
                cx="29"
                cy="14"
                fill={
                  Array [
                    0,
                    4294949576,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                r="7"
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
              <RNSVGCircle
                cx="14"
                cy="14"
                fill={
                  Array [
                    0,
                    4294967295,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                r="7"
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
              <RNSVGCircle
                cx="29"
                cy="29"
                fill={
                  Array [
                    0,
                    4287683839,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                r="7"
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
              <RNSVGCircle
                cx="14"
                cy="29"
                fill={
                  Array [
                    0,
                    4294934287,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                r="7"
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 0,
            "flexDirection": "row",
          },
          Object {
            "width": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "flexGrow": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexGrow": 1,
              "flexShrink": 0,
              "paddingHorizontal": 14,
            }
          }
        >
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "flexShrink": 0,
                        "fontFamily": "GTGuardianTitlepiece-Bold",
                        "fontSize": 24,
                        "lineHeight": 26,
                      },
                      Array [
                        Object {
                          "color": "#ffffff",
                          "marginTop": -2,
                        },
                        undefined,
                        Object {},
                      ],
                    ]
                  }
                >
                  Thursday
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "flexShrink": 0,
                        "fontFamily": "GTGuardianTitlepiece-Bold",
                        "fontSize": 24,
                        "lineHeight": 26,
                      },
                      Array [
                        Object {
                          "color": "#ffffff",
                          "marginTop": -2,
                        },
                        Object {
                          "color": "#ffe500",
                        },
                        Object {},
                      ],
                    ]
                  }
                >
                  25 June
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
        <View
          style={
            Object {
              "paddingRight": 14,
            }
          }
        >
          <View
            accessibilityHint="More issues"
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 999,
                    "height": 42,
                    "justifyContent": "center",
                    "paddingHorizontal": 21,
                  },
                  Object {
                    "backgroundColor": "#ffe500",
                  },
                  Object {
                    "alignItems": "center",
                    "aspectRatio": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 0,
                    "width": 42,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "fontFamily": "GuardianIcons-Regular",
                      "fontSize": 20,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "color": "#121212",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
